controllers/installation_controller.go

```diff
@@ -937,6 +937,23 @@ func (r *InstallationReconciler) DisableOldInstallations(ctx context.Context, it
 	}
 }
 
+// ReadClusterConfigSpecFromSecret reads the cluster config from the secret pointed by spec.ConfigSecret
+// if it is set. This overrides the default configuration from spec.Config.
+func (r *InstallationReconciler) ReadClusterConfigSpecFromSecret(ctx context.Context, in *v1beta1.Installation) error {
+	if in.Spec.ConfigSecret == nil {
+		return nil
+	}
+	var secret corev1.Secret
+	nsn := types.NamespacedName{Namespace: in.Spec.ConfigSecret.Namespace, Name: in.Spec.ConfigSecret.Name}
+	if err := r.Get(ctx, nsn, &secret); err != nil {
+		return fmt.Errorf("failed to get config secret: %w", err)
+	}
+	if err := in.Spec.ParseConfigSpecFromSecret(secret); err != nil {
```

Comment:
Just to make sure, this works even if there are fields in the secret that are not yet known to operator?

Reply:
Yes, it does work but that got me thinking I haven't added a test case for that. Here: https://github.com/replicatedhq/embedded-cluster-kinds/pull/22